tests/syntax_tests/data/parsing/errors/typeDef/expected/inlineRecord.res.txt

```diff
@@ -32,17 +20,20 @@
      │ nder(props)
   20 │ 
 
-  An inline record type declaration is only allowed in a variant constructor's declaration
+  An inline record type declaration is only allowed in a variant constructor's declaration or nested inside of a record type declaration
 
 type nonrec entity =
   | Director 
   | Student of {
   name: string ;
   reportCard: < passing: bool  ;score: int   >  } 
-type nonrec user =
+type user.address = {
+  street: string ;
+  country: string }[@@res.inlineRecordDefinition ]
+and user =
   {
   name: string ;
-  address: < street: string  ;country: string   >  }
+  address: ((user.address)[@res.inlineRecordReference ]) }
```

Comment:
Why use both `res.inlineRecordReference` and `res.inlineRecordDefinition`?
Presumably if one can find `user.address`, one can also determine that it is an inline record definition. Or perhaps there's some convenience given by this.

Would you add some parsing and printing tests too.

Reply:
Thinking about parsing and printing -- this representation makes you search for ids such as `user.address`.
Wondering whether going directly with a new AST representation could actually simplify this.
The compilation to usual type definitions could perhaps be done after parsing/printing, as for JSX.
On the fence on whether it's simpler to investigate this sooner or later.

Reply:
Reason for having an explicit attribute is because I didn't want to hijack having `.` in the name meaning that it's an inline record. But perhaps that is OK, given that it's still an illegal identifier, and it only takes effect if there's also an inlined type named that available.

I think an actual AST representation makes sense as well. Not sure how to best structure it though.

Reply:
Ahh I see what you mean now. Perf wise it shouldn't be much of a hit - it'll only attempt lookup in the cases when there are actual inlined types, and here we could safely look for a `.` in the name before attempting as well. I'll see if I can change it to work like that real quick.

Reply:
@cristianoc good catch, was able to get rid of `res.inlineRecordReference` here: https://github.com/rescript-lang/rescript/pull/7241/commits/82fc9c8bf706ff8f4cd59e52f5a318eed1fa92c5